src/workers/usernameChanged.ts

```diff
@@ -0,0 +1,58 @@
+import { notifyCommentsUpdate } from './../common/pubsub';
+import { CommentMention } from './../entity/CommentMention';
+import { messageToJson, Worker } from './worker';
+
+interface Data {
+  userId: string;
+  oldUsername: string;
+  newUsername: string;
+}
+
+const limit = parseInt(process.env.COMMENT_BATCH_UPDATE_LIMIT || '50');
+
+const worker: Worker = {
+  subscription: 'username-changed-api',
+  handler: async (message, con, logger): Promise<void> => {
+    const { userId, oldUsername, newUsername }: Data = messageToJson(message);
+    try {
+      const commentMentions = await con
+        .getRepository(CommentMention)
+        .find({ mentionedUserId: userId });
+      const commentIds = commentMentions.map((mention) => mention.commentId);
+      const slices = commentIds.length / limit + 1;
+      const batches = [];
+      for (let slice = 0; slice < slices; slice++) {
+        const start = slice * limit;
+        const end = (slice + 1) * limit;
```

Comment:
What happens if `end` is bigger than array size?

Reply:
Tried it in a console yesterday and it doesn't result in an error but fetches what's left instead.